css/css-logical/logical-longhand-properties.html

```diff
@@ -0,0 +1,262 @@
+  },
+  "margin-*": {
+    type: "length",
+  },
+  "padding-*": {
+    type: "length",
+  },
+  "inset-*": {
+    type: "length",
+    prerequisites: {"position": "relative"},
+  },
+};
+
+// Values to use while testing
+const testValues = {
+  "length": ["1px", "2px", "3px", "4px", "5px"],
+  "color": ["rgb(1, 1, 1)", "rgb(2, 2, 2)", "rgb(3, 3, 3)", "rgb(4, 4, 4)", "rgb(5, 5, 5)"],
+  "border-style": ["solid", "dashed", "dotted", "double", "groove"],
+};
+
+// Six unique overall writing modes for property-mapping purposes.
+const writingModes = [
+  {
+    styles: [
+      {"writing-mode": "horizontal-tb", "direction": "ltr"},
+    ],
+    blockStart: "top", blockEnd: "bottom", inlineStart: "left", inlineEnd: "right",
+    block: "vertical", inline: "horizontal" },
+  {
+    styles: [
+      {"writing-mode": "horizontal-tb", "direction": "rtl"},
+    ],
+    blockStart: "top", blockEnd: "bottom", inlineStart: "right", inlineEnd: "left",
+    block: "vertical", inline: "horizontal" },
+  {
+    styles: [
+      {"writing-mode": "vertical-rl", "direction": "rtl"},
+      {"writing-mode": "sideways-rl", "direction": "rtl"},
+    ],
+    blockStart: "right", blockEnd: "left", inlineStart: "bottom", inlineEnd: "top",
+    block: "horizontal", inline: "vertical" },
+  {
+    styles: [
+      {"writing-mode": "vertical-rl", "direction": "ltr"},
+      {"writing-mode": "sideways-rl", "direction": "ltr"},
+    ],
+    blockStart: "right", blockEnd: "left", inlineStart: "top", inlineEnd: "bottom",
+    block: "horizontal", inline: "vertical" },
+  {
+    styles: [
+      {"writing-mode": "vertical-lr", "direction": "rtl"},
+      {"writing-mode": "sideways-lr", "direction": "ltr"},
+    ],
+    blockStart: "left", blockEnd: "right", inlineStart: "bottom", inlineEnd: "top",
+    block: "horizontal", inline: "vertical" },
+  {
+    styles: [
+      {"writing-mode": "vertical-lr", "direction": "ltr"},
+      {"writing-mode": "sideways-lr", "direction": "rtl"},
+    ],
+    blockStart: "left", blockEnd: "right", inlineStart: "top", inlineEnd: "bottom",
+    block: "horizontal", inline: "vertical" },
+];
+
+// Creates a group of physical and logical box properties, such as
+//
+// { physical: {
+//     left: "margin-left", right: "margin-right",
+//     top: "margin-top", bottom: "margin-bottom",
+//   }, logical: {
+//     inlineStart: "margin-inline-start", inlineEnd: "margin-inline-end",
+//     blockStart: "margin-block-start", blockEnd: "margin-block-end",
+//   }, type: "length", prerequisites: "..." }
+//
+// where the `type` is a key from the `testValues` object and the `prerequisites`
+// is a declaration including prerequisites for all four physical properties.
+function createBoxPropertyGroup(property, descriptor) {
+  const logical = {};
+  const physical = {};
+  for (const logicalSide of ["inline-start", "inline-end", "block-start", "block-end"]) {
+    const camelCase = logicalSide.replace(/-(.)/g, (match, $1) => $1.toUpperCase());
```

Comment:
Why you need this line? Wouldn't it be the same just using `logicalSide` later instead of `camelCase`.

Reply:
Logical properties are converted into physical using the properties from the `writingModes` items. If I use `logicalSide` here, then `writingModes` should have properties like `"block-start"` instead of `blockStart`. The lines will be longer due to the quotes and hyphen, but OK.